### ISSUE TITLE
Thermostat temperature is wrong within HA

### ISSUE THREAD

**Issue by royduin** [jbouwh/incomfort-client#40]
It's been a while since [I worked with it](https://github.com/royduin/home-assistant-incomfort) since the contributions to HA directly. But recently I saw some behavior change; when I set the temperature from the thermostat. It doesn't get reflected within HA. Is that intended?

Example; I've set the temperature to 21 but I don't see that anywhere in HA. It stays 14:
<img width="304" alt="image" src="https://github.com/user-attachments/assets/dca1384a-7fc0-41e6-999a-42f478b794ab" />

From the Incomfort web interface:
<img width="508" alt="image" src="https://github.com/user-attachments/assets/36b915cf-80ce-4c54-ad06-99041e35d6e5" />

**Comment by jbouwh** [comment]
Not sure what you mean, the setpoint override is the value that is used by HA, it can take a minute before the setpoint it self is reported back from the bouler to the gateway.

**Comment by jbouwh** [comment]
Btw in ha core 2024.12.4 there is a fix, in some cases, where the heater is not the first defined entry, changing the setpoint did not work.

**Comment by royduin** [comment]
I'm running 2024.12.4. But when I set my thermostat to 21, I expect to see 21 in HA. That's how it always worked. When changing the temperature from HA the setpoint override is set. Not sure why that's changed? Currently I can't even find the setpoint of 21 within HA.

**Comment by jbouwh** [comment]
Right, did it work with 2024.12.3 en before?

**Comment by royduin** [comment]
The issue is there for longer, not sure when it occurred. When scrolling through the changes: https://github.com/home-assistant/core/commits/dev/homeassistant/components/incomfort, this sounds like the issue: https://github.com/home-assistant/core/pull/119528. What you're describing there was sometimes the issue but you just have to wait until it's updated within the gateway.

**Comment by jbouwh** [comment]
right, not that #119528 only affected the climates state after changing the override.
Is your heater responding when you set an override?

**Comment by jbouwh** [comment]
When I update my Intergas roomthermostat, it will take a minute or 2 before HA updates the setpoint, but it works.

**Comment by royduin** [comment]
When everything is set to 14 degrees Celsius:
```
{"nodenr": 229,
"ch_temp_lsb": 24,
"ch_temp_msb": 20,
"tap_temp_lsb": 216,
"tap_temp_msb": 15,
"ch_pressure_lsb": 151,
"ch_pressure_msb": 0,
"room_temp_1_lsb": 40,
"room_temp_1_msb": 7,
"room_temp_set_1_lsb": 120,
"room_temp_set_1_msb": 5,
"room_temp_2_lsb": 255,
"room_temp_2_msb": 127,
"room_temp_set_2_lsb": 255,
"room_temp_set_2_msb": 127,
"displ_code": 126,
"IO": 0,
"serial_year": 16,
"serial_month": 5,
"serial_line": 16,
"serial_sn1": 0,
"serial_sn2": 93,
"serial_sn3": 83 ,
"room_set_ovr_1_msb": 5,
"room_set_ovr_1_lsb": 120,
"room_set_ovr_2_msb": 0,
"room_set_ovr_2_lsb": 0,
"rf_message_rssi": 32,
"rfstatus_cntr": 0}
```

I see:

<img width="423" alt="Screenshot 2024-12-22 at 14 56 13" src="https://github.com/user-attachments/assets/5d993bdd-855c-466c-a53e-43209f540d64" />

---

When I change the temperature to 20 degrees Celsius with the fysical thermostat on the wall:

```
{"nodenr": 229,
"ch_temp_lsb": 24,
"ch_temp_msb": 20,
"tap_temp_lsb": 216,
"tap_temp_msb": 15,
"ch_pressure_lsb": 151,
"ch_pressure_msb": 0,
"room_temp_1_lsb": 40,
"room_temp_1_msb": 7,
"room_temp_set_1_lsb": 208,
"room_temp_set_1_msb": 7,
"room_temp_2_lsb": 255,
"room_temp_2_msb": 127,
"room_temp_set_2_lsb": 255,
"room_temp_set_2_msb": 127,
"displ_code": 126,
"IO": 0,
"serial_year": 16,
"serial_month": 5,
"serial_line": 16,
"serial_sn1": 0,
"serial_sn2": 93,
"serial_sn3": 83 ,
"room_set_ovr_1_msb": 5,
"room_set_ovr_1_lsb": 120,
"room_set_ovr_2_msb": 0,
"room_set_ovr_2_lsb": 0,
"rf_message_rssi": 32,
"rfstatus_cntr": 0}
```

I still see the same:

<img width="423" alt="Screenshot 2024-12-22 at 14 56 13" src="https://github.com/user-attachments/assets/5d993bdd-855c-466c-a53e-43209f540d64" />

---

Only when I change the temperature from HA to 21 degrees Celsius:

```
{"nodenr": 229,
"ch_temp_lsb": 175,
"ch_temp_msb": 12,
"tap_temp_lsb": 191,
"tap_temp_msb": 13,
"ch_pressure_lsb": 176,
"ch_pressure_msb": 0,
"room_temp_1_lsb": 30,
"room_temp_1_msb": 7,
"room_temp_set_1_lsb": 208,
"room_temp_set_1_msb": 7,
"room_temp_2_lsb": 255,
"room_temp_2_msb": 127,
"room_temp_set_2_lsb": 255,
"room_temp_set_2_msb": 127,
"displ_code": 0,
"IO": 10,
"serial_year": 16,
"serial_month": 5,
"serial_line": 16,
"serial_sn1": 0,
"serial_sn2": 93,
"serial_sn3": 83 ,
"room_set_ovr_1_msb": 8,
"room_set_ovr_1_lsb": 52,
"room_set_ovr_2_msb": 0,
"room_set_ovr_2_lsb": 0,
"rf_message_rssi": 32,
"rfstatus_cntr": 0}
```

I see the temperature:

<img width="479" alt="Screenshot 2024-12-22 at 15 02 31" src="https://github.com/user-attachments/assets/8bdeb7f6-7414-47bd-a3aa-bdccdc5b8326" />

---

When I change the temperature back to 20 with the fysical thermostat, HA still says 21. So it seems like the setpoint override is used to display the current temperature instead of the setpoint.

<img width="540" alt="image" src="https://github.com/user-attachments/assets/ffaa662f-611d-40d7-a369-be40a7581c23" />

```
{"nodenr": 229,
"ch_temp_lsb": 23,
"ch_temp_msb": 21,
"tap_temp_lsb": 3,
"tap_temp_msb": 15,
"ch_pressure_lsb": 194,
"ch_pressure_msb": 0,
"room_temp_1_lsb": 35,
"room_temp_1_msb": 7,
"room_temp_set_1_lsb": 208,
"room_temp_set_1_msb": 7,
"room_temp_2_lsb": 255,
"room_temp_2_msb": 127,
"room_temp_set_2_lsb": 255,
"room_temp_set_2_msb": 127,
"displ_code": 0,
"IO": 10,
"serial_year": 16,
"serial_month": 5,
"serial_line": 16,
"serial_sn1": 0,
"serial_sn2": 93,
"serial_sn3": 83 ,
"room_set_ovr_1_msb": 8,
"room_set_ovr_1_lsb": 52,
"room_set_ovr_2_msb": 0,
"room_set_ovr_2_lsb": 0,
"rf_message_rssi": 32,
"rfstatus_cntr": 0}
```

<img width="479" alt="Screenshot 2024-12-22 at 15 02 31" src="https://github.com/user-attachments/assets/8bdeb7f6-7414-47bd-a3aa-bdccdc5b8326" />

**Comment by jbouwh** [comment]
When I change the setpoint with the wall thermostat I see the set point is changed (override =0) after a few. After a few minutes the adjusted temperature is updated in HA. When I change the temp from HA (=override) I see the thermostat confirming the new setpoint. After some time the setpoint is updated to be in sync with the override. If this is still not correct you can find me on discord (same alias).

**Comment by jbouwh** [comment]
I have updated the client `inclient.py` a bit. Did you try using it?

Example to set a setpoint to 15 degC (heater 0, room 0):
`python inclient.py 192.168.1.x -u admin -p password -H 0 -R 0 -t 15`

To show the current status (json):
`python inclient.py 192.168.1.x -u admin -p password -H 0 -R 0`

**Comment by jbouwh** [comment]
I'lll close this issue. Note that the library only support setting an override. When that is done (via HA) it takes some time before the setpoint is updated. To avoid indicating the wrong info, HA shows the override as temp set  (if set) or else the setpoint if override is 0. When the thermostat is changes fysically, but there was an override set as well, it might take some time before the temperature setting is updated.

**Comment by royduin** [comment]
Currently I've set the temperature with the physical thermostat on the wall to 20.5, the client outputs:
```
{'display_code': <DisplayCode.STANDBY: 126>, 'display_text': 'standby', 'fault_code': None, 'is_burning': False, 'is_failed': False, 'is_pumping': False, 'is_tapping': False, 'heater_temp': 38.33, 'tap_temp': 33.2, 'pressure': 1.63, 'serial_no': '1605g009383', 'nodenr': 229, 'rf_message_rssi': 30, 'rfstatus_cntr': 0, 'room_1': {'room_temp': 20.89, 'setpoint': 20.5, 'override': 14.0}}
```

And within HA I see:
<img width="436" alt="image" src="https://github.com/user-attachments/assets/c1ac0162-4e54-42d8-ad89-6c55ef233d78" />

Also within the graph I don't see the actual setpoint:
<img width="171" alt="image" src="https://github.com/user-attachments/assets/b1f84e1d-eea8-48c9-a0e5-40806c2fb179" />

So it seems like within HA the `override` is used instead of the `setpoint`.

> To avoid indicating the wrong info, HA shows the override as temp set

So `setpoint` is wrong? Why? I think using the `override` is wrong. That's how it worked for a long time. Still not sure why and where that's changed.

Can't the `setpoint` be used instead of the `override` in HA?

**Comment by jbouwh** [comment]
That is weird. Override should be reset after the setpoint changed. Override is indeed used if it is set.

**Comment by jbouwh** [comment]
> Can't the setpoint be used instead of the override in HA?

Yes, that was the case earlier, but has been changed. In that case would an override not been shown until the setpoint had been synced.

**Comment by jbouwh** [comment]
So: Yes, if setpoint would be used instead to override, then HA would be updated faster when the thermostat is changed manually. The current code is using the override because, HA can only set the override temperature via the gateway. In that case it will take a few minutes for the setpoint to sync from the override, but during that period HA will show the correct setpoint.
When you change HA, and while waiting manually change the thermostat, it will take some time before the override value is reset. After that HA will indicate the correct temperature again. From the data it is hard to tell what the setpoint will become while there is an override set. The choice is to show the override to offer the best user response in the HA UI.

**Comment by royduin** [comment]
> Override should be reset after the setpoint changed.

> The override is reset to 0.

That's not the case with mine.

> Yes, that was the case earlier, but has been changed.

Can that be changed back or made configurable? Not sure which models do that but with mine the `setpoint` should always be used.

**Comment by jbouwh** [comment]
This is a gateway propery. May be your gateway needs a reboot? Via de web ui you can change overrides. This is what the library uses to change the setpoint via setting an override. Not sure what you mean.

If you change the override, the thermostat should respond and show the new temperature. After a while setpoint will sync with override (not be `0` yet). If you change the setpoint manually from the thermostat, then the override will be set to `0`.

**Comment by royduin** [comment]
Maybe there is a difference between gateways? I've this one:

![image](https://github.com/user-attachments/assets/2d46c00f-b15b-4aa5-aee7-e3ed19187417)

The override doesn't change to 0 ever. I'm also unable to set the override to 0 from the web ui.

**Comment by jbouwh** [comment]
I probably have about the same hardware. What firmware version are you on.

**Comment by royduin** [comment]
Web interface version: 1.1
Firmware version: IC3-ICN-V1.12
Build date: Mar 16 2016 00:08:00

**Comment by jbouwh** [comment]
Ah okay:
I cleary have a newer version of the gateway.

Web interface version: |   | 1.4
-- | -- | --
Firmware version: |   | IC3-2-ICN-V1.30
Build date: |   | May 09 2022 00:08:00

May be an update is possible?

I'll see if I can build in an option to allow the setpoint attibute to be the used for the target temperature.

**Comment by royduin** [comment]
It seems like it's not possible to update; https://gathering.tweakers.net/forum/list_message/55522241#55522241, in 2019 I contacted Intergas and got the same answer.

> De LAN2RF gateways kunnen niet worden voorzien van een andere firmware. Updaten is dus niet mogelijk, daar is de hardware niet voor geschikt.

An option to choose between `setpoint` and `override` would be nice! 

Is that a big thing? With some tips maybe I could manage it myself, but not sure if it will meet all standards/rules. I don't want to burden you with this. https://github.com/royduin/home-assistant-incomfort was the best I could build back then, but a lot changed since.

**Comment by jbouwh** [comment]
Opened a PR to add the option in HA core.

**Comment by royduin** [comment]
Wow, that's fast! Thanks!! 🚀🚀

**Comment by jbouwh** [comment]
Thank you for persisting in addressing the problem.

**Comment by jbouwh** [comment]
Closing. The new options should be available with HA Core 2025.2

**Comment by royduin** [comment]
Awesome! 🚀 looking forward to it 😎 thanks for helping out!
